src/AutoRest.CSharp/Common/Output/PostProcessing/PostProcessor.cs

```diff
@@ -0,0 +1,294 @@
+    public async Task<Project> InternalizeAsync(Project project)
+    {
+        var compilation = await project.GetCompilationAsync();
+        if (compilation == null)
+            return project;
+
+        // first get all the declared symbols
+        var definitions = await GetTypeSymbolsAsync(compilation, project, true);
+        // build the reference map
+        var referenceMap = await new ReferenceMapBuilder(compilation, project, HasDiscriminator).BuildPublicReferenceMapAsync(definitions.DeclaredSymbols, definitions.DeclaredNodesCache);
+        // get the root symbols
+        var rootSymbols = GetRootSymbols(project, definitions);
+        // traverse all the root and recursively add all the things we met
+        var publicSymbols = VisitSymbolsFromRootAsync(rootSymbols, referenceMap);
+
+        var symbolsToInternalize = definitions.DeclaredSymbols.Except(publicSymbols);
+
+        var nodesToInternalize = new List<BaseTypeDeclarationSyntax>();
+        foreach (var symbol in symbolsToInternalize)
+        {
+            nodesToInternalize.AddRange(definitions.DeclaredNodesCache[symbol]);
+        }
+
+        foreach (var model in nodesToInternalize)
+        {
+            project = MarkInternal(project, model);
+        }
+
+        return project;
+    }
+
+    /// <summary>
+    /// This method removes the no-referenced types from the <paramref name="project"/>. It will do this job in the following steps:
+    /// 1. This method will read all the defined types in the given <paramref name="project"/>, and build a cache for those symbols
+    /// 2. Build a reference map for those symbols (including non-public usage)
+    /// 3. Finds all the root symbols, please override the <see cref="IsRootDocument(Document)"/> to control which document you would like to include
+    /// 4. Visit all the symbols starting from the root symbols following the reference map to get all unvisited symbols
+    /// 5. Remove the definition of the unvisited symbols in step 4
+    /// </summary>
+    /// <param name="project">The project to process</param>
+    /// <returns>The processed <see cref="Project"/>. <see cref="Project"/> is immutable, therefore this should usually be a new instance </returns>
+    public async Task<Project> RemoveAsync(Project project)
+    {
+        var compilation = await project.GetCompilationAsync();
+        if (compilation == null)
+            return project;
+
+        // find all the declarations, including non-public declared
+        var definitions = await GetTypeSymbolsAsync(compilation, project, false);
+        // build reference map
+        var referenceMap = await new ReferenceMapBuilder(compilation, project, HasDiscriminator).BuildAllReferenceMapAsync(definitions.DeclaredSymbols, definitions.DocumentsCache);
+        // get root nodes
+        var rootNodes = GetRootSymbols(project, definitions);
```

Comment:
To be consistent with the codes above.
```suggestion
        var rootSymbols = GetRootSymbols(project, definitions);
```